lib/models/ObjectLockConfiguration.js

```diff
@@ -0,0 +1,183 @@
+                'request xml does not contain Days or Years');
+            return validTime;
+        }
+        const timeValue = Number.parseInt(dr[timeType][0], 10);
+        if (Number.isNaN(timeValue)) {
+            validTime.error = errors.MalformedXML.customizeDescription(
+                'request xml does not contain valid retention period');
+            return validTime;
+        }
+        validTime.timeType = timeType.toLowerCase();
+        validTime.timeValue = timeValue;
+        return validTime;
+    }
+
+    /**
+     * Check that object lock configuration is valid
+     * @return {object} - contains error if parsing failed
+     */
+    _parseObjectLockConfig() {
+        const validConfig = {};
+        if (!this._parsedXml || this._parsedXml === '') {
+            validConfig.error = errors.MalformedXML.customizeDescription(
+                'request xml is undefined or empty');
+            return validConfig;
+        }
+        const objectLockConfig = this._parsedXml.ObjectLockConfiguration;
+        if (!objectLockConfig && objectLockConfig !== '') {
+            validConfig.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include ObjectLockConfiguration');
+            return validConfig;
+        }
+        const objectLockEnabled = objectLockConfig.ObjectLockEnabled;
+        if (!objectLockEnabled || objectLockEnabled[0] !== 'Enabled') {
+            validConfig.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include valid ObjectLockEnabled');
+            return validConfig;
+        }
+        const ruleArray = objectLockConfig.Rule;
+        if (ruleArray.length > 1) {
+            validConfig.error = errors.MalformedXML.customizeDescription(
+                'request xml contains more than one rule');
+            return validConfig;
+        }
+
+        const drArray = ruleArray[0].DefaultRetention;
+        if (!drArray || !drArray[0] || drArray[0] === '') {
+            validConfig.error = errors.MalformedXML.customizeDescription(
+                'Rule request xml does not contain DefaultRetention');
+            return validConfig;
+        }
+        if (!drArray[0].Mode || (!drArray[0].Days && !drArray[0].Years)) {
+            validConfig.error = errors.MalformedXML.customizeDescription(
+                'DefaultRetention request xml does not contain Mode or ' +
+                'retention period (Days or Years)');
+            return validConfig;
+        }
+        const validMode = this._parseMode(drArray[0].Mode);
+        if (validMode.error) {
+            validConfig.error = validMode.error;
+            return validConfig;
+        }
+        const validTime = this._parseTime(drArray[0]);
+        if (validTime.error) {
+            validConfig.error = validTime.error;
+            return validConfig;
+        }
+
+        this._config.rule = {};
```

Comment:
I think in addition to the `rule`, `objectLockEnabled` property could be added to the config as well (as it seems to be a part of the `objectLockConfiguration` type per AWS specs), but also, as I wrote the getBucketObjectLock in a way that it returns the response in the following format anyway (by adding ObjectLockEnabled property if object lock is enabled):
```
{
    "ObjectLockConfiguration": {
        "ObjectLockEnabled": "Enabled",
        "Rule": {
            "DefaultRetention": {
                "Mode": "COMPLIANCE",
                "Days": 50
            }
        }
    }
}
```
so I couldn't be sure if it would necessarily needed to be added, think we should be good either way.

Reply:
Nevermind about this one; we’re actually already storing this exactly as `ObjectLockEnabled` in the bucket info already. :)